04.ls/ls.rb

```diff
@@ -55,16 +57,18 @@ def split_files(files)
 }.freeze
 
 def show_long_format(path, file_names)
-  path_to_stat = file_names.to_h { |file_name| [File.join(path, file_name), File.lstat(File.join(path, file_name))] }
+  absolute_path_list = file_names.map { |file_name| File.join(path, file_name) }
+  stats_by_path = absolute_path_list.to_h { |absolute_path| [absolute_path, File.lstat(absolute_path)] }
+  maxlength_by_section = fetch_maxlength(stats_by_path.values)
```

Comment:
```suggestion
  max_length_by_section = fetch_max_length(stats_by_path.values)
```

単語を分けた方がいいですね。

あとハッシュを表現するのに `<value>_by_<key>` とするのは明示的でいいのですが、単純に複数形にするだけでも良かったりします。

```suggestion
  max_lengths = fetch_max_length(stats_by_path.values)
```

これといって正解はないのでどっちがいいとは断言できませんが、可読性を下げない範囲で極力シンプルな名前を検討してみてください〜。

Reply:
`maxlength_by_section`などは、ハッシュの形を明示するために`max`と`length`を合わせていました。
ハッシュの構成が重要な`stats_by_path`以外は、`~s`といった複数形に統一しました。